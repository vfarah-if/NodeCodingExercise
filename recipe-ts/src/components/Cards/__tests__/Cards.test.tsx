import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {
	DefaultLayout,
	TwoColumnLayout
} from '../Cards.stories';

describe('Cards', () => {
	it('should create a default four column cards list', () => {
		const { container } = render(
			<DefaultLayout columnCount={DefaultLayout.args?.columnCount} />
		);

		expect(container).toMatchSnapshot();
	});

	it('should create a default 2 column cards list', () => {
		const { container } = render(
			<TwoColumnLayout columnCount={TwoColumnLayout.args?.columnCount} />
		);

		expect(container).toMatchSnapshot();
	});
});
