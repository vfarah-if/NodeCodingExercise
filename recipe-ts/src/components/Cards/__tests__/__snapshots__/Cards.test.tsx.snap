// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should create a default 2 column cards list 1`] = `
<div>
  <ul
    style="display: grid; grid-template-columns: repeat(2, 1fr);"
  >
    <li
      class="card"
    >
      <h1>
        1
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        2
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        3
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        4
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        5
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        6
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        7
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        8
      </h1>
    </li>
  </ul>
</div>
`;

exports[`Cards should create a default four column cards list 1`] = `
<div>
  <ul
    style="display: grid; grid-template-columns: repeat(4, 1fr);"
  >
    <li
      class="card"
    >
      <h1>
        1
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        2
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        3
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        4
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        5
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        6
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        7
      </h1>
    </li>
    <li
      class="card"
    >
      <h1>
        8
      </h1>
    </li>
  </ul>
</div>
`;
